gtda/images/filtrations.py

```diff
@@ -117,8 +118,8 @@ def fit(self, X, y=None):
         X = check_array(X, allow_nd=True)
         self.n_dimensions_ = X.ndim - 1
         if (self.n_dimensions_ < 2) or (self.n_dimensions_ > 3):
-            warn(f"Input of `fit` contains arrays of dimension "
-                 f"{self.n_dimensions_}.")
+            raise ValueError(f"Input of `fit` contains arrays of dimension "
```

Comment:
This change and the ones below make a lot of the codes in `test_common` break. Would it be too bad to allow 1-dimensional arrays? After all, we now allow collections of 1-dimensional arrays as input to `CubicalPersistence`.

Reply:
They all break I believe...
I will make the change!